fsclient/client.go

```diff
@@ -0,0 +1,115 @@
+package fsclient
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"log"
+	"mime/multipart"
+	"net/http"
+	url2 "net/url"
+)
+
+type FileServerClient struct {
+	adr string
+	username string
+	password string
```

Comment:
both potential users of this api will have the password encrypted in memory. please add an encryption struct that will be used to decrypt the password on demand and accept in the constructor below